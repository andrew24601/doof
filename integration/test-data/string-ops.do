// Test string operations
// Tests: ADD_STRING, LENGTH_STRING, EQ_STRING

function main(): int {
    let str1: string = "Hello";
    let str2: string = " World";
    let str3: string = "Hello";
    
    let concat: string = str1 + str2;  // ADD_STRING
    let len1: int = str1.length;       // LENGTH_STRING
    let len2: int = concat.length;     // LENGTH_STRING
    let equal: bool = str1 == str3;    // EQ_STRING (true)
    let not_equal: bool = str1 == str2; // EQ_STRING (false)
    
    println(concat);    // Should print "Hello World"
    println(len1);      // Should print 5
    println(len2);      // Should print 11
    println(equal);     // Should print true
    println(not_equal); // Should print false
    
    return 0;
}
