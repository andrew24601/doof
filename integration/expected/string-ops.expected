Hello World
5
11
true
false
